File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.swift
import Foundation

func solution(_ n:Int, _ m:Int) -> [Int] {
    let gcd = getGCD(n ,m)
    let lcm = (n * m) / gcd
    
    return [gcd, lcm]
}

func getGCD(_ x:Int , _ y: Int) -> Int {
    // a > b
    let a = max(x, y)
    let b = min(x, y)
    
    return (a % b != 0) ? getGCD(b, a % b) : b
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(int a, int b) {
    int c = a % b;
    while(c != 0) {
        a = b;
        b = c;
        c = a % b;
    }
    return b;
}

int lcm(int a, int b) {
    return (a * b) / gcd(a, b);
}


vector<int> solution(int n, int m) {
    vector<int> answer;

    int gcdVal = n > m ? gcd(n, m) : gcd(m, n);
    int lcmVal = n > m ? lcm(n, m) : lcm(m, n);

    answer.push_back(gcdVal);
    answer.push_back(lcmVal);

    return answer;
}